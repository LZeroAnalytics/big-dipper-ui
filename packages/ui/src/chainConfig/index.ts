import chainJson from '@/chain.json';
import type { ChainConfig } from '@/chainConfig/types';

const chainTypes = [
  typeof process !== 'undefined' && process.env ? process.env.NEXT_PUBLIC_CHAIN_TYPE : undefined,
];

function chainConfig() {
  /* Setting the basePath, chainType, chains, and settings variables. */
  const chainType = chainTypes.find((c) => !!c)?.toLowerCase() || 'mainnet';
  const { chains, ...settings } = chainJson;
  let chain = chains.find((c) => c.chainType?.toLowerCase() === chainType);
  if (!chain && chainType !== 'testnet') {
    chain = chains.find((c) => c.chainType?.toLowerCase() === 'testnet');
  }

  /* If the chainType is not found, it will use the first chain in the array. */
  if (!chain) [chain] = chains;
  if (!chain) throw new Error(`Config not found for CHAIN_NAME ${chainJson.chainName}`);

  const basePath = (
    (typeof process !== 'undefined' && process.env ? process.env.BASE_PATH : undefined) ||
    `${`/${settings.chainName}`}`
  ).replace(/^(\/|\/base)$/, '');

  /* Merging the settings and chain objects and processing environment variables. */
  const processedChain = {
    ...chain,
    endpoints: {
      ...chain.endpoints,
      graphql: chain.endpoints?.graphql?.replace(
        '${NEXT_PUBLIC_GRAPHQL_URL}',
        (typeof process !== 'undefined' && process.env
          ? process.env.NEXT_PUBLIC_GRAPHQL_URL
          : undefined) || chain.endpoints.graphql
      ),
      graphqlWebsocket: chain.endpoints?.graphqlWebsocket?.replace(
        '${NEXT_PUBLIC_GRAPHQL_WS}',
        (typeof process !== 'undefined' && process.env
          ? process.env.NEXT_PUBLIC_GRAPHQL_WS
          : undefined) || chain.endpoints.graphqlWebsocket
      ),
      publicRpcWebsocket: chain.endpoints?.publicRpcWebsocket?.replace(
        '${NEXT_PUBLIC_RPC_WEBSOCKET}',
        (typeof process !== 'undefined' && process.env
          ? process.env.NEXT_PUBLIC_RPC_WEBSOCKET
          : undefined) || chain.endpoints.publicRpcWebsocket
      ),
    },
  };

  return {
    ...settings,
    basePath,
    ...processedChain,
  } as unknown as ChainConfig;
}

export default chainConfig;
