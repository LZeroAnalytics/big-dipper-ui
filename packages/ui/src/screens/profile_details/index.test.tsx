import { useMemo } from 'react';
import renderer from 'react-test-renderer';
import chainConfig from '@/chainConfig';
import ProfileDetails from '@/screens/profile_details';
import MockTheme from '@/tests/mocks/MockTheme';
import wait from '@/tests/utils/wait';

const { prefix } = chainConfig();

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    query: {
      dtag: '@dtag',
    },
    pathname: '/@dtag',
    replace: jest.fn(() => '/'),
    push: jest.fn(() => '/@dtag'),
  }),
}));
jest.mock('@/components/layout', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Layout" {...props} />
));
jest.mock('@/components/load_and_exist', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="LoadAndExist" {...props} />
));

jest.mock(
  '@/screens/profile_details/components/connections',
  () => (props: JSX.IntrinsicElements['div']) => <div id="Connections" {...props} />
);

// ==================================
// unit tests
// ==================================
describe('screen: ProfileDetails', () => {
  it('matches snapshot', async () => {
    let component: renderer.ReactTestRenderer | undefined;

    renderer.act(() => {
      component = renderer.create(
        <MockTheme>
          <ProfileDetails />
        </MockTheme>
      );
    });
    await wait(renderer.act);

    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
