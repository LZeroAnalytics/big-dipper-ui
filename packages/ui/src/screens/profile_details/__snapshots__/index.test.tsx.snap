// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: ProfileDetails matches snapshot 1`] = `
.emotion-0 {
  padding: 16px;
  display: grid;
  grid-template-rows: auto;
  gap: 8px;
}

@media (min-width: 1280px) {
  .emotion-0 {
    padding: 16px 24px;
  }
}

.emotion-0 a {
  color: #4092CD;
}

@media (min-width:1280px) {
  .emotion-0 {
    gap: 16px;
  }
}

<div
  id="Layout"
  navTitle="profileDetails"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <span
      className="emotion-0"
    >

      <div
        data={
          [
            {
              "creationTime": "2021-10-06T00:10:45.761731",
              "identifier": "desmostest",
              "network": "desmos",
            },
          ]
        }
        id="Connections"
      />
    </span>
  </div>
</div>
`;
